### DISCUSSION TITLE
Support of multiple modes.

### DISCUSSION THREAD

**Discussion by Krolken** [pwitab/iec62056-21#22] [Ideas]
We might want to be able to support mutiple readout modes.

**Comment by emontnemery** [comment]
How about splitting out the specifics for each mode to subclasses of `Iec6205621Client`, the same way it's done for the transports?

**Reply by Krolken** [reply]
I am trying to move most of my communication libraries to a sans-io approach. So I would like to make a better distinction between the messages sent, the baudrate needed and the order the messages can be sent.

Mode A:
* Request Message - Expect an Identification Message
* Enter Programming mode - ACK, NACK, or EXIT
* Read, Write, EXIT - READ-response, Write-response, EXIT.

Mode B:
* Request Message - Expect an Identification Message - Then a DataMessage
*  Enter Programming mode - ACK, NACK, or EXIT
* Read, Write, EXIT - READ-response, Write-response, EXIT.

Mode C
* Request Message - Expect an Identification Message
* AckWithDataReadout
* * Expect a DataMessage for readout
* AckWithProgrammingMode
* * Expect PasswordChallange
* * P1 or P2 password
* * ACK, NACK or EXIT
* * Read, Write, EXIT - READ-response, Write-response, EXIT.

* AckWithManufacturespecific
* * Leave it up to someone else to deal with that.

Mode D:
* Just listen to full readout. I am planning this feature for P1 port server.

Mode E:
* Needed for my DLMS library to move a meter to DLMS. But messages should be defined here for better reuse.
